cpp/inline/games/stochastic_nim/PerfectPlayer.inl

```diff
@@ -0,0 +1,58 @@
+#include <games/stochastic_nim/PerfectPlayer.hpp>
+
+namespace stochastic_nim {
+
+inline PerfectPlayer::ActionResponse PerfectPlayer::get_action_response(
+    const State& state, const ActionMask& valid_actions) {
+  util::release_assert(state.current_mode == kPlayerMode,
+                       "PerfectPlayer does not support chance mode");
+  return strategy_->get_optimal_action(state.stones_left);
+}
+
+PerfectStrategy::PerfectStrategy() {
+  for (int i = 0; i < stochastic_nim::kStartingStones + 1; i++) {
+    state_values_[i] = -1.0;
+    optimal_actions_[i] = -1;
+  }
+
+  state_values_[0] = 1.0;
+  state_values_[1] = 0.8;
+  state_values_[2] = 0.5;
+  state_values_[3] = 0.0;
+  optimal_actions_[1] = 1;
+  optimal_actions_[2] = 2;
+  optimal_actions_[3] = 3;
```

Comment:
I would not hard-code here. Use the values defined in `Constants.hpp`.

Reply:
In PerfertStrategy, the optimal_actions are defined as "the number of stones taken" to make the math easier to reason about and overall more straightforward to work in DP. In Constants.hpp. kTake1 = 0, kTake2 = 1 and kTake3 = 2, which refers to the i-th action. I realized the conceptual difference and thought it might be okay to encapsulate this into the get_optimal_action() which is to be interfaced from the outside world. Should I change it to follow the definition of "the i-th action"?